Mathlib/FieldTheory/SeparableClosure.lean

```diff
@@ -0,0 +1,386 @@
+  one_mem' := (map_one (algebraMap F E)) ▸ separable_algebraMap F E 1
+  add_mem' := separable_add F E
+  zero_mem' := (map_zero (algebraMap F E)) ▸ separable_algebraMap F E 0
+  algebraMap_mem' := separable_algebraMap F E
+  inv_mem' := separable_inv F E
+
+/-- An element is contained in the (relative) separable closure of `E / F` if and only if
+it is a separable element. -/
+theorem mem_separableClosure_iff {x : E} :
+    x ∈ separableClosure F E ↔ (minpoly F x).Separable := Iff.rfl
+
+/-- If `i` is an `F`-algebra homomorphism from `E` to `K`, then `i x` is contained in
+`separableClosure F K` if and only if `x` is contained in `separableClosure F E`. -/
+theorem map_mem_separableClosure_iff (i : E →ₐ[F] K) {x : E} :
+    i x ∈ separableClosure F K ↔ x ∈ separableClosure F E := by
+  simp_rw [mem_separableClosure_iff, minpoly.algHom_eq i i.injective]
+
+/-- If `i` is an `F`-algebra homomorphism from `E` to `K`, then `separableClosure F E` is equal to
+the preimage of `separableClosure F K` under the map `i`. -/
+theorem separableClosure.eq_comap_of_algHom (i : E →ₐ[F] K) :
+    separableClosure F E = (separableClosure F K).comap i := by
+  ext x
+  exact (map_mem_separableClosure_iff F E K i).symm
+
+/-- If `i` is an `F`-algebra homomorphism from `E` to `K`, then `separableClosure F K` contains
+the image of `separableClosure F E` under the map `i`. -/
+theorem separableClosure.map_le_of_algHom (i : E →ₐ[F] K) :
+    (separableClosure F E).map i ≤ separableClosure F K := fun x hx ↦ by
+  change x ∈ (_ : Set K) at hx
+  rw [coe_map, Set.mem_image] at hx
+  obtain ⟨y, hy, rfl⟩ := hx
+  exact (map_mem_separableClosure_iff F E K i).2 hy
+
+/-- If `K / E / F` is a field extension tower, such that `K / E` has no non-trivial separable
+subextensions (when `K / E` is algebraic, this means that it is purely inseparable),
+then `separableClosure F K` is equal to `separableClosure F E`. -/
+theorem separableClosure.eq_map_of_separableClosure_eq_bot [Algebra E K] [IsScalarTower F E K]
+    (h : separableClosure E K = ⊥) :
+    separableClosure F K = (separableClosure F E).map (IsScalarTower.toAlgHom F E K) := by
+  refine le_antisymm (fun x hx ↦ ?_) (map_le_of_algHom F E K _)
+  rw [mem_separableClosure_iff] at hx
+  obtain ⟨y, hy⟩ := mem_bot.1 <| h ▸ (mem_separableClosure_iff E K).2 (hx.map_minpoly E)
+  rw [← hy, minpoly.algebraMap_eq (algebraMap E K).injective, ← mem_separableClosure_iff] at hx
+  exact ⟨y, hx, hy⟩
+
+/-- If `i` is an `F`-algebra isomorphism of `E` and `K`, then `separableClosure F K` is equal to
+the image of `separableClosure F E` under the map `i`. -/
+theorem separableClosure.eq_map_of_algEquiv (i : E ≃ₐ[F] K) :
+    separableClosure F K = (separableClosure F E).map i.toAlgHom := by
+  refine le_antisymm (fun x hx ↦ ?_) (map_le_of_algHom F E K i.toAlgHom)
+  change x ∈ (_ : Set K)
+  rw [coe_map, Set.mem_image]
+  exact ⟨i.symm.toAlgHom x,
+    (map_mem_separableClosure_iff F K E i.symm.toAlgHom).2 hx, i.apply_symm_apply x⟩
```

Comment:
defeq manipulation before `exact` is almost always unnecessary. `.toAlgHom` are unnecessary too.
```suggestion
    separableClosure F K = (separableClosure F E).map i := by
  refine le_antisymm (fun x hx ↦ ?_) (map_le_of_algHom F E K i)
  exact ⟨i.symm x, (map_mem_separableClosure_iff F K E i.symm).2 hx, i.right_inv x⟩
```
(`i.symm x` can be omitted too)